pilot/test/util/diff.go

```diff
@@ -110,12 +113,39 @@ func StripVersion(yaml []byte) []byte {
 func RefreshGoldenFile(content []byte, goldenFile string, t *testing.T) {
 	if Refresh() {
 		t.Logf("Refreshing golden file %s", goldenFile)
-		if err := ioutil.WriteFile(goldenFile, content, 0644); err != nil {
+		// nolint: gocritic
+		if err := AtomicWrite(goldenFile, content, 0644); err != nil {
 			t.Errorf(err.Error())
 		}
 	}
 }
 
+// Write atomically by writing to a temporary file in the same directory then renaming
+func AtomicWrite(path string, data []byte, mode os.FileMode) (err error) {
```

Comment:
Reuse the existing `AtomicWrite` function used in Istio CNI. It's better, as it doesn't leave the tmp file behind in case of error, etc.
https://github.com/istio/istio/blob/master/cni/pkg/install-cni/pkg/util/util.go#L45

Reply:
Or was it intentional to leave the tmp file for debugging?